### PR TITLE
EDU-894 Create a search client for Wikimedia Commons files

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-894

I stopped here after having a working search client, because I think from here on we can directly implement a new tab on the resource picker without the need for mocking up a preliminary UI.

### CODE REVIEW

File: src/api-clients/wikimedia-commons-api-client.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      gsrnamespace: '*',
+      prop: 'info|imageinfo',
+      inprop: 'url',
+      iiprop: 'url|size|mime',
+      iiurlheight: thumbnailHeight,
+      origin: '*'
+    });
+
+    return this.httpClient
+      .get(`${WIKIMEDIA_COMMONS_API_URL}?${queryString}`, { responseType: 'json' })
```
Comment (adrianaluchian) [inline src/api-clients/wikimedia-commons-api-client.js:39]: I'm thinking we should url encode the `queryString` part due to `searchText`, or?

Comment (ahelmberger) [reply src/api-clients/wikimedia-commons-api-client.js:39]: `urlUtils.composeQueryString` returns an encoded, ready to use query string.

File: src/components/pages/tests.js [hunk 2]
```diff
@@ -74,6 +76,19 @@ function Tests({ PageTemplate }) {
...
 
+  // WikimediaCommonsApiClient
+  const [wcacQuery, setWcacQuery] = useState('');
+  const [wcacResult, setWcacResult] = useState('');
+  const wcacApiClient = useService(WikimediaCommonsApiClient);
+  const [wcacFileTypes, setWcacFileTypes] = useState(Object.values(FILE_TYPE));
+  const handleWcacSearchClick = async () => {
+    const data = await wcacApiClient.queryMediaFiles({ searchText: wcacQuery, fileTypes: wcacFileTypes });
+    setWcacResult(JSON.stringify(data, null, 2));
+    // For future reference: `data.continue.gsroffset` (if set) gives us the next offset to load even more results:
```
Comment (adrianaluchian) [inline src/components/pages/tests.js:87]: Nice!

File: src/api-clients/wikimedia-commons-api-client.js (hunk 1 above)
Comment (adrianaluchian) [reply src/api-clients/wikimedia-commons-api-client.js:39]: Indeed it does 😬

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**